projects/devices/tdk/protos/Mpu-9250.proto

```diff
@@ -34,6 +34,9 @@ PROTO Mpu-9250 [
       Compass {
         name %<= '"' + fields.name.value + ' compass"' >%
       }
+      InertialUnit {
+        name %<= '"' + fields.name.value + ' inertial_unit"' >%
+      }
```

Comment:
We use in [rosbot_ros2_firmware](https://github.com/husarion/rosbot_ros2_firmware) Mpu-9250 with give us the orientation of the robot.
The best way to add to the ROSbots the IMU would be to implement  `webots_imu_hardware_interface' with `ros2_control` and then use `imu_broadcaster` from `ros2_controllers`.

Reply:
I see, but since in webots this device gives ground truth information, is not very realistic. In webots_ros2 the IMU is also [modeled](https://github.com/cyberbotics/webots_ros2/wiki/References-Devices#imu-device) using the three devices (gyro, accelerometer & compass). Perhaps a middle ground could be to add a new boolean parameter to the PROTO that adds/removes the presence of this inertial unit, that gives the users the choice to either use the ground truth information or the more realistic approach. This can be achieved easily with [javascript templating](https://cyberbotics.com/doc/reference/javascript-procedural-proto) in the PROTO. You just need to add `# template language: javascript` to the header of the PROTO and wrap the device:
```

...

field SFBool includeInertialUnit FALSE

...

%< if (fields.includeInertialUnit.value) { >%
      InertialUnit {
        name %<= '"' + fields.name.value + ' inertial_unit"' >%
      }
%< } >%
```

Reply:
@ad-daniel 
Maybe could I add Inertial Unit to the Rosbot.proto and RosbotXl.proto?

Reply:
In theory yes, but I think the other approach is better since the `Mpu-9250` is what is available in the real robot. That way it's up to the user then which to use, whether the ground truth provided by the inertial unit or the sensor fused information.